File: go/pangram/pangram.go
package pangram

const testVersion = 1

func IsPangram(s string) bool {
	charCount := make(map[rune]int)
	for i := 0; i < 26; i++ {
		charCount[rune('a'+i)] = 0
	}

	for _, c := range s {
		if c >= 65 && c <= 90 {
			c = c + 32
		}
		if charCount[c] == 0 {
			charCount[c] += 1
		}
	}

	for _, count := range charCount {
		if count == 0 {
			return false
		}
	}
	return true
}

File: python/pangram/pangram.py
def is_pangram(sentence):
    if sentence == "":
        return False
    h = {}
    for c in range(ord('a'), ord('z')):
        h[chr(c)] = 0
    for c in sentence.lower():
        if h.get(c) == 0:
            h[c] += 1
    for c, count in h.items():
        if count == 0:
            return False
    return True
